python/pyarrow/_flight.pyx

```diff
@@ -855,10 +888,14 @@ cdef class FlightInfo(_Weakrefable):
             the descriptor for this flight.
         endpoints : list of FlightEndpoint
             a list of endpoints where this flight is available.
-        total_records : int
-            the total records in this flight, or -1 if unknown
-        total_bytes : int
-            the total bytes in this flight, or -1 if unknown
+        total_records : int optional, default -1
+            the total records in this flight, or -1 if unknown.
+        total_bytes : int optional, default -1
+            the total bytes in this flight, or -1 if unknown.
```

Comment:
`None` would be a preferrable default for a Python API, IMHO.

Reply:
Me too, but the attributes will have to return `-1` (because of the C++ backend), so that would be a bit inconsistent. @kou your thoughts?

Reply:
But that might be a breaking change.

Reply:
None isn't a real value for these. (It's also inconsistent with the underlying protobuf, which doesn't really have optional values, and would default to 0 for a truly unspecified value.)

Reply:
I agree with lidavidm.

Reply:
I'm not sure what a "real value" is? `None` is the standard placeholder in Python for missing or unknown data. I agree that it would be a slight compatibility break to start returning `None` instead of -1, but that's for the better IMHO.

Reply:
The underlying structure does not admit a null value. It is Protobuf, there are no true null values (*this has changed slightly in recent versions but we do not use that codegen option for this structure). 

Let's be clear here, though. Is this for input, or output? I would be -1 to ever returning `None` for this attribute. If it's merely to accept it as an input on construction that might be OK, but it would be misleading to people when it doesn't round trip.

Reply:
This does not affect the underlying structure. The Python `FlightEnpoint` can take `None` values and translate to `-1` for the underlying C++ class. The getter methods can translate those `-1` back to None, so this Python class feels pythonic.

Reply:
> The getter methods can translate those -1 back to None, so this Python class feels pythonic.

This is what I would object to. 

> The Python FlightEnpoint can take None values and translate to -1 for the underlying C++ class.

I don't mind this though it should be clearly documented (as the underlying Protobuf does not take None).

Reply:
To summarize, the constructor allows for `-1` and `None`, the properties in both cases return `-1`.